server/chat/backend/agent/tools/notion/workspace.py

```
@@ -0,0 +1,574 @@
+    user_id: Optional[str] = None,
+    session_id: Optional[str] = None,
+) -> str:
+    """Upload a file to Notion via the file_uploads API (single- or multi-part)."""
+    _ = session_id
+
+    if not file_path_or_url or not file_path_or_url.strip():
+        return notion_tool_error("file_path_or_url is required", code="bad_input")
+
+    try:
+        data, src_filename = _fetch_bytes(file_path_or_url.strip())
+    except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)